### ISSUE TITLE
doc_markdown suggests backticks for text that is not an identifier

### ISSUE THREAD

**Issue by leighmcculloch** [rust-lang/rust-clippy#13097]
### Summary

The doc_markdown lint suggested backticks for text that is not an identifier anywhere in the code file or in the workspace.

### Lint Name

doc_markdown

### Reproducer

I tried this code:

```rust
#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, clap::ValueEnum)]
pub enum OutputFormat {
    /// Colorful, human-oriented console output
    Pretty,
    /// Human-oriented console output without colors
    Plain,
    /// JSONified console output
    Json,
}
```

I saw this happen:

```
error: item in documentation is missing backticks
   --> cmd\soroban-cli\src\commands\events.rs:131:9
    |
131 |     /// JSONified console output
    |         ^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
help: try
    |
131 |     /// `JSONified` console output
    |         ~~~~~~~~~~~
```

Ref: https://github.com/stellar/stellar-cli/actions/runs/9933470510/job/27436261859?pr=1453

I expected to see this happen:

No warning, as the text "JSONified" doesn't refer to an identifier.

### Version

```text
1.79.0-x86_64-pc-windows-msvc installed - rustc 1.79.0 (129f3b996 2024-06-10)
```

**Comment by GuillaumeGomez** [comment]
Taking a look.
